lib/logstash/inputs/snmptrap.rb

```diff
@@ -48,46 +52,36 @@ class LogStash::Inputs::Snmptrap < LogStash::Inputs::Base
   config :community, :validate => :array, :default => 'public'
 
   # directory of YAML MIB maps  (same format ruby-snmp uses)
-  config :yamlmibdir, :validate => :string
-
-  # Defines a target field for placing fields.
-  # If this setting is omitted, data gets stored at the root (top level) of the event.
-  # The target is only relevant while decoding data into a new event.
-  config :target, :validate => :field_reference
-
-  BASE_MIB_PATH = ::File.join(__FILE__, '..', '..', '..', 'mibs')
+  config :yamlmibdir, :validate => :string, :deprecated => "Use `mib_paths` instead."
 
   # These MIBs were automatically added by ruby-snmp when no @yamlmibdir was provided.
-  DEFAULT_MIB_PATHS = [
-    ::File.join(BASE_MIB_PATH, 'ietf', 'SNMPv2-SMI.dic'),
-    ::File.join(BASE_MIB_PATH, 'ietf', 'SNMPv2-MIB.dic'),
-    ::File.join(BASE_MIB_PATH, 'ietf', 'IF-MIB.dic'),
-    ::File.join(BASE_MIB_PATH, 'ietf', 'IP-MIB.dic'),
-    ::File.join(BASE_MIB_PATH, 'ietf', 'TCP-MIB.dic'),
-    ::File.join(BASE_MIB_PATH, 'ietf', 'UDP-MIB.dic')
+  MIB_DEFAULT_PATHS = [
+    ::File.join(MIB_BASE_PATH, 'ietf', 'SNMPv2-SMI.dic'),
+    ::File.join(MIB_BASE_PATH, 'ietf', 'SNMPv2-MIB.dic'),
+    ::File.join(MIB_BASE_PATH, 'ietf', 'IF-MIB.dic'),
+    ::File.join(MIB_BASE_PATH, 'ietf', 'IP-MIB.dic'),
+    ::File.join(MIB_BASE_PATH, 'ietf', 'TCP-MIB.dic'),
+    ::File.join(MIB_BASE_PATH, 'ietf', 'UDP-MIB.dic')
   ].map { |path| ::File.expand_path(path) }
 
   def initialize(params={})
     super(params)
 
+    setup_deprecated_params!
     @host_ip_field = ecs_select[disabled: 'host', v1: '[host][ip]']
   end
 
   def register
-    mib_manager = MibManager.new(RubySnmpOidFieldMapper.new)
-
-    if @yamlmibdir
-      logger.info("Loading user-provided MIB files", :path => @yamlmibdir)
-      mib_manager.add(@yamlmibdir)
-    else
-      logger.info("Loading default MIB files", :path => DEFAULT_MIB_PATHS)
-      DEFAULT_MIB_PATHS.each do |path|
+    mib_manager = build_mib_manager!
+
+    if !@mib_paths && !@use_provided_mibs
```

Comment:
This means that with `use_provided_mibs => true`, the AND becomes false, so the default MIB files aren't loaded.
IIRC we should default `use_provided_mibs` to true so that they're loaded by default, and skip only if the user provides another path for them.

Reply:
The current traps behaviour is:
- If no @mib_paths is provided, `ruby-snmp` loads some default [modules](https://github.com/hallidave/ruby-snmp/blob/master/lib/snmp/options.rb#L46) internally, which are not all the IETF MIBs.
- If an @mib_paths is provided, it only loads the provided files, and not the internally default modules.

This condition was written this way to keep it backward compatible. If no MIBs are provided through `use_provided_mibs` or `mib_paths`, it only loads the same default modules ruby-snmp loads. 

The mib_paths and use_provided_mibs conditions are implemented on the [build_mib_manager!](https://github.com/logstash-plugins/logstash-integration-snmp/pull/6/files/85bf005d61d7897eaf9e16992549f339fcb4e6f5#diff-62a57a3743817ac82c2d9d0ad4a249028b9ee3799af5e45d80633237e84a4877R82).  This extra condition is specific for the traps plugin.

That said, do you think we should change the traps `use_provided_mibs` default to `true`? My concern is that OIDs that weren't being resolved before will change, and users will start seeing event fields with the names instead of OIDs dotted strings.

Reply:
My comment came from a test where I compared the default behavior of snmptrap input when receiving the following trap:

```
snmptrap -v 2c -c public 127.0.0.1:1062 '' SNMPv2-MIB::coldStart SNMPv2-MIB::sysName.0 s "MyDevice"
```

In the current Logstash 8.13.x the loaded mibs were able to translate the `1.3.6.1.6.3.1.1.5.1` reference as `coldStart`:

```ruby
                         "host" => "127.0.0.1",
                      "message" => "#<SNMP::SNMPv2_Trap:0x5e56cf33 @varbind_list=[#<SNMP::VarBind:0x8b5c64f @name=[1.3.6.1.2.1.1.3.0], @value=#<SNMP::TimeTicks:0x55460b61 @value=0>>, #<SNMP::VarBind:0x3d0169ad @name=[1.3.6.1.6.3.1.1.4.1.0], @value=[1.3.6.1.6.3.1.1.5.1]>, #<SNMP::VarBind:0x44762606 @name=[1.3.6.1.2.1.1.5.0], @value=\"MyDevice\">], @request_id=991566823, @error_index=0, @error_status=0, @source_ip=\"127.0.0.1\">",
    "SNMPv2-MIB::snmpTrapOID.0" => "SNMPv2-MIB::coldStart",
      "SNMPv2-MIB::sysUpTime.0" => "00:00:00.00",
        "SNMPv2-MIB::sysName.0" => "MyDevice",
                     "@version" => "1",
                   "@timestamp" => 2024-04-15T13:30:27.935624Z
```

Where the integration's snmptrap didn't translate it:

```
/tmp/logstash-8.13.1 29s
❯ bin/logstash -e "input { snmptrap {  } } output { stdout { codec => rubydebug { } } }" --pipeline.ecs_compatibility=disabled                  
{
                         "host" => "127.0.0.1",
                      "message" => "{\"error_index\"=>0, \"variable_bindings\"=>{\"1.3.6.1.6.3.1.1.4.1.0\"=>\"1.3.6.1.6.3.1.1.5.1\", \"1.3.6.1.2.1.1.5.0\"=>\"MyDevice\", \"1.3.6.1.2.1.1.3.0\"=>0}, \"error_status\"=>0, \"type\"=>\"TRAP\", \"error_status_text\"=>\"Success\", \"community\"=>\"public\", \"version\"=>\"2c\", \"request_id\"=>1738241066}",
      "SNMPv2-MIB::sysUpTime.0" => 0,
        "SNMPv2-MIB::sysName.0" => "MyDevice",
                     "@version" => "1",
                   "@timestamp" => 2024-04-15T13:30:35.891716Z,
    "SNMPv2-MIB::snmpTrapOID.0" => "1.3.6.1.6.3.1.1.5.1"
}

```

Reply:
Actually now that I notice, it doesn't translate the reference regardless of the value of `use_provided_mibs`

Reply:
Oh I see, the reference value translation is not implemented on the MIB manager. It works for the current traps plugin because `ruby-snmp` does it automatically under the hood. Do you think we should do the same here? The `snmp` input currently does not support it as well so the support could be added for both plugins.

Reply:
It seems like a quick win (not breaking BC) and a desired SNMP input feature (https://github.com/logstash-plugins/logstash-input-snmp/issues/35). If you don't mind, I can implement it on this PR! :)

Reply:
I've pushed the changes (https://github.com/logstash-plugins/logstash-integration-snmp/pull/6/commits/de0a8e8796589c63c6ca4ef38dcd22140dea2898), please let me know your thoughts.